### PR TITLE
Feat/add reencrypt keystore

### PR DESCRIPTION
**Pull request by shanefontaine**
* Add ability to `reencrypt` keystores
* Only major change is that a custom file path needs to be explicitly passed in with `—path` instead of `args[1]`. I did this because I didn’t think it was possible for a user to know that `args[1]` should be the path, but happy to revert to what it was if you think otherwise.
* DRYed some logic that is now re-used b/c of new command

### CODE REVIEW

**Review by miguelmota — CHANGES_REQUESTED** [review]

**Comment by miguelmota** [comment]
The reencryption works great

Some minor bugs

Issue 1:
Expected: Throw error after entering invalid current passphrase and not proceed to enter new passphrase
Actual: No error was thrown when entering invalid current passphrase and it proceeded to prompt for new passphrase

DeepinScreenshot_select-area_20220108175957

Issue 2:
Expected: Screen to not clear after reencrypting new passphrase.
Actual: Screen cleared after rencrypting new passphrase. 

DeepinScreenshot_select-area_20220108180110

I think the screen should only clear when showing the mnemonic. The screen should not clear when reencrypting. It should not show the 'press enter to exit' prompt.  It should only show the 'reencryption is complete' log and behave similar to decrypt command.

File: packages/hop-node/src/cli/keystores.ts [hunk 1]
```diff
@@ -12,31 +12,46 @@ import { hopArt } from './shared/art'
...
   .option('--override [boolean]', 'Override existing keystore if it exists.', parseBool)
   .option('--private-key <private-key>', 'The private key to encrypt.', parseString)
   .action(actionHandler(main))
 
 async function main (source: any) {
-  let { config, override, args, pass: passphrase, output, privateKey } = source
+  let { override, args, pass: passphrase, path: keystoreFilePath, privateKey } = source
   const action = args[0]
-  output = output || defaultKeystoreFilePath
+  const actionOptions = [
```
[the remark this answers is not in the document]
Comment (shanefontaine) [reply packages/hop-node/src/cli/keystores.ts]: Resolved with ef6b0199ea112dd3eeddab2e1ac0ea575adb2fda

**Comment by shanefontaine** [comment]
I agree with all comments. They've been resolved in ef6b0199ea112dd3eeddab2e1ac0ea575adb2fda. Docs have been updated.

**Comment by miguelmota** [comment]
@shanefontaine the updates work great 👍

**Review by miguelmota — APPROVED** [review]

**Pull request merged**